src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,387 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.integration
+
+/-!
+
+# Vector valued measures
+
+This file defines vector valued measures, which are σ-additive functions from a set to a add monoid
+`M` such that it maps the empty set and non-measurable sets to zero. In the case
+that `M = ℝ`, we called the vector measure a signed measure and write `signed_measure α`.
+Similarly, when `M = ℂ`, we call the measure a complex measure and write `complex_measure α`.
+
+## Main definitions
+
+* `vector_measure` is a vector valued, σ-additive function that maps the empty
+  and non-measurable set to zero.
+
+## Implementation notes
+
+We require all non-measurable sets to be mapped to zero in order for the extensionality lemma
+to only compare the underlying functions for measurable sets.
+
+We use `has_sum` instead of `tsum` in the definition of vector measures in comparison to `measure`
+since this provides summablity.
+
+## Tags
+
+vector measure, signed measure, complex measure
+-/
+
+noncomputable theory
+
+open_locale classical big_operators nnreal ennreal
+
+namespace measure_theory
+
+variables {α β : Type*} [measurable_space α]
+
+/-- A vector measure on a measurable space `α` is a σ-additive `M`-valued function (for some `M`
+an add monoid) such that the empty set and non-measurable sets are mapped to zero. -/
+structure vector_measure (α : Type*) [measurable_space α]
+  (M : Type*) [add_comm_monoid M] [topological_space M] :=
+(measure_of : set α → M)
+(empty : measure_of ∅ = 0)
+(not_measurable ⦃i : set α⦄ : ¬ measurable_set i → measure_of i = 0)
+(m_Union ⦃f : ℕ → set α⦄ :
+  (∀ i, measurable_set (f i)) → pairwise (disjoint on f) →
+  has_sum (λ i, measure_of (f i)) (measure_of (⋃ i, f i)))
```

Comment:
The convention used by `monoid_hom` etc is to name these fields with a trailing `'`, and then provide copies of them that use `⇑v` instead of `v.measure_of` that omit the trailing `'`.

I note that the `measure_theory` library doesn't do this, but I think it's worth considering anyway - it's easiet for a user to pick between `vector_measure.empty'` and `vector_measure.empty` ("well, I probably want the unprimed one unless I know better") than it is to pick between `vector_measure.empty` and `vector_measure.of_empty`.

Reply:
I think I prefer the `'` convention but it might be better to keep in line with the convention from `measure_theory`? I'm not sure what is better here since using a different convention might be confusing

Reply:
`measure_theory` predates the convention, that's why it doesn't use it (also it doesn't use dot notation as much as it could). In new code, using the new convention with `'` is probably better.